File: openmp/hello.cc
#include <iostream>
#include <omp.h>

using namespace std;

void say_hello(int my_thread_num, int total_threads)
{
   cout << "Hello I am thread " << my_thread_num << " among " << total_threads
        << " threads\n";
}

int main()
{
   cout << "I am the master, and I am alone\n";

   #pragma omp parallel
   {
      say_hello(omp_get_thread_num(), omp_get_num_threads());
   }
   return 0;
}

File: openmp/hello.f90
program main
   use omp_lib

   print*,"I am the master, and I am alone"

!$OMP PARALLEL
   call say_hello(omp_get_thread_num(), omp_get_num_threads())
!$OMP END PARALLEL

end program main

subroutine say_hello(my_thread_num, total_threads)
   integer :: my_thread_num, total_threads
   print*,"Hello, I am thread ",my_thread_num,"among ",total_threads,"threads"
end subroutine say_hello
